packages/backend/src/models/CatalogModel/CatalogModel.ts

```diff
@@ -114,15 +116,46 @@ export class CatalogModel {
                                     .where('project_uuid', projectUuid)
                                     .delete();
 
-                                const inserts = await this.database
-                                    .batchInsert(
-                                        CatalogTableName,
-                                        catalogInserts,
+                                const results = await trx(CatalogTableName)
+                                    .insert(
```

Comment:
be aware , IIRC in the past we had some problems when inserting a lot of queries, I believe there is a limit , and can throw errors.